tests/set.rs

```diff
@@ -1,3 +1,4 @@
+#![cfg(not(miri))] // FIXME: takes too long
```

Comment:
Might be better to add this to the test, not the entire file?

Reply:
I thought about that, but then we need to add it to the extern crate and use statements as well to silence the warnings about unused stuff. There is only one test in the file, so I thought that for the time being this was the cleanest thing to do, but if that changes, I agree with you that we should run the other tests under miri.